articles/container-components-in-react/index.md

```diff
@@ -0,0 +1,195 @@
+
+Before we start building this app, let's have look at how the folder structure containing all the files are represented. You do not compulsorily need to go with this type of structure if you have already have one that is suitable for you. The goal is just to be able to under the whole flow of the project.
+
+```
+src
+ |-- App.js
+ |-- bookshelf.js
+ |-- components
+           |---- Books.js
+           |---- Search.js
+```
+
+Let’s start by thinking about the function(s) that the search component would perform. A search component should be a able to perform basic tasks like searching and filtering.
+
+To keep things simple, let's use an array of objects as the data to search from, instead of using an API.
+
+Let's create an array of objects containing simple information about books that will be in this component:
+
+```js
+// bookshelf.js
+export const books = [
+  {
+    name: "Eloquent JavaScript",
+    author: "Marijn Haverbeke",
+  },
+  {
+    name: "CSS The Definitive Guide",
+    author: "Eric A. Meyer & Estelle Weyl",
+  },
+  {
+    name: "Automate the boring stuff with Python",
+    author: "Al Sweigart",
+  },
+  {
+    name: "Make: JavaScript Robotics",
+    author: "Donovan Buck, Bryan Hughes et al",
+  },
+];
+```
+
+The data in the `bookshelf.js` file will serve as the information that would be mapped onto the `App` component, so the search component can be able to filter the list of authors once the user starts typing into the input field.
+
+Let's create the components needed in this app below and I'll explain how everything works.
+
+The `Search` component presents the UI (i.e. It renders an input element where users can filter the books in the list.) to the user. It makes use of React props to pass data down to other components such as the author’s page container component.
+
+```js
+// search.js
+import React from “react”
+
+const Search = ({ val, onSearch, ...props }) => {
+  return (
+    <input
+	  type="text"
+	  value={val}
+	  placeholder="Search for any book"
+	  onChange={onSearch}
+	  {...props}
+	/>
+  );
+};
+
+export default Search
+```
+
+The `Books` component performs almost the same function as the search component. Most people refer to this type of components as **"dumb", "stateless" or "presentational"** components because they're only concerned with how the user interface of the application is being rendred or displayed to the user. They do not always have state, instead they use props to share data in the application component tree.
+
+Take a look at how the `bookShelf` component is being imported into the authors container and how it is being used in the `Books` component.
+
+```js
+// books.js
+const Books = ({ books }) => {
+  return (
+  {books.map((bookShelf, index) => {
+    return (
+	  <ul key={index}>
+	    <li>Book: {bookShelf.bookName}</li>
+		<li>Author: {bookShelf.author}</li>
+	  </ul>
+    )
+  })}
+  )
+}
```

Comment:
```suggestion
const Books = ({ books }) => {
  return (
  {books.map((book, index) => {
    return (
	  <ul key={index}>
	    <li>Book: {book.bookName}</li>
		<li>Author: {book.author}</li>
	  </ul>
    )
  })}
  )
}
```

Reply:
Since you've already named the file bookshelf, rename these as `book` to prevent any sort of confusion.

Reply:
Ok...

That sounds good